lib/model/card/action/card_action.dart

```diff
@@ -17,7 +17,9 @@ abstract class CardAction {
   void performAction(GameState state) {
     state.playerHand.deductResources(cost);
     performActionImpl(state);
-    state.nextCard();
+    if (state.result == null) {
+      state.nextCard();
+    }
```

Comment:
Now that we've got this, we could extend the tests in `end_game_action_test` to also check that nextCard() wasn't called (`currentCard` should be unchanged)

Reply:
Fair. Done.